File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, h: str, n: str) -> int:
        ans=""
        if h==n:
            return 0
        s=""    
        for i in range(len(h)):
            s+=h[i]
            if(s==n):
                return i-len(n)+1
            if(len(s)>=len(n) and s!=n):
                s=s[1:len(s)]
        return -1            

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string h, string n) {
        string s = "";
        if (h == n)
            return 0;
        for (int i = 0; i < h.size(); i++) {
            s += h[i];
            if (s == n)
                return i - n.size() + 1;
            if (s.size() >= n.size() && s != n) {
                s.erase(s.begin());
            }
        }
        return -1;
    }
};